components/rewards/RewardsFooter.js

```diff
@@ -1,51 +1,72 @@
 import { FontAwesome5 } from '@expo/vector-icons';
-import PropTypes from 'prop-types';
 import React from 'react';
-import { View } from 'react-native';
+import { AsyncStorage, View } from 'react-native';
 import Colors from '../../constants/Colors';
+import RecordIds from '../../constants/RecordIds';
+import { getCustomersById } from '../../lib/airtable/request';
 import { Subhead } from '../BaseComponents';
 /**
  * @prop
  * */
 
-function RewardsFooter({ customer, isGuest }) {
-  return (
-    <View
-      style={{
-        width: '100%',
-        flexDirection: 'row',
-        justifyContent: 'space-between',
-        padding: 16,
-      }}>
-      {customer && (
-        <View
-          style={{
-            flexDirection: 'row',
-            alignItems: 'center',
-          }}>
-          <FontAwesome5 name="star" solid size={16} color={Colors.lightest} />
-          <Subhead style={{ paddingLeft: 8 }} color={Colors.lightest}>
-            {isGuest
-              ? 'Learn about Healthy Rewards >'
-              : `${customer.points} points`}
-          </Subhead>
-        </View>
-      )}
-      {customer && !isGuest && (
-        <Subhead color={Colors.lightest}>Your next reward &gt;</Subhead>
-      )}
-    </View>
-  );
-}
-
-RewardsFooter.propTypes = {
-  customer: PropTypes.object,
-  isGuest: PropTypes.bool,
-};
+export default class RewardsFooter extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      customer: null,
+      transactions: [],
+      isLoading: true,
+      isGuest: false,
+    };
+  }
+  // Load customer record & transactions
+  async componentDidMount() {
+    const customerId = await AsyncStorage.getItem('customerId');
+    const isGuest = customerId === RecordIds.guestCustomerId;
+    try {
+      const customer = await getCustomersById(customerId);
+      this.setState({
+        isGuest,
+        customer,
+        isLoading: false,
+      });
+    } catch (err) {
+      console.error(err);
+      logErrorToSentry({
+        screen: 'RewardsFooter',
+        action: 'componentDidMount',
+        error: err,
+      });
+    }
+  }
 
-RewardsFooter.defaultProps = {
-  customer: null,
-  isGuest: true,
-};
-
-export default RewardsFooter;
+  render() {
+    return (
+      <View
+        style={{
+          width: '100%',
+          flexDirection: 'row',
+          justifyContent: 'space-between',
+          padding: 16,
+        }}>
+        {this.state.customer && (
+          <View
+            style={{
+              flexDirection: 'row',
+              alignItems: 'center',
+            }}>
+            <FontAwesome5 name="star" solid size={16} color={Colors.lightest} />
+            <Subhead style={{ paddingLeft: 8 }} color={Colors.lightest}>
+              {this.state.isGuest
+                ? 'Learn about Healthy Rewards >'
+                : `${this.state.customer.points} points`}
```

Comment:
@thumn mentioned a while ago that this could be really confusing since this displays total points, and the Rewards page shows progress toward the next reward instead. It should probably be `rewardPointValue - pointsToNext` like on the rewards progress bar in `RewardsHome`. I followed up with @12aschen for updated designs on this.

Reply:
changed!